### ISSUE TITLE
The maxFile and maxsize property isn't working.

### ISSUE THREAD

**Issue by swateek** [winstonjs/winston-daily-rotate-file#12]
To test my code, I am rotating the log file every minute.

```
var logger = new (winston.Logger)({
  transports: [
    new (winston.transports.Console)({ json: true, timestamp: true, colorize: true }),
    new (require('winston-daily-rotate-file'))({ dirname: '/'+__dirname , filename: 'debug.log', datePattern: '.yyyy-MM-ddTHH-mm', maxFiles: 2, maxsize: 104857600 })
  ],
  exceptionHandlers: [
    new (winston.transports.Console)({ json: true, timestamp: true, colorize: true }),
    new winston.transports.File({ filename: __dirname + '/exceptions.log', json: true })
  ],
  exitOnError: false
});

```

My sample code for logging is here - 

```
var recursive = function () {
  logger.debug('log to file');
  logger.info("127.0.0.1 - there's no place like home");
  logger.warn("127.0.0.1 - there's no place like home");
  logger.error("127.0.0.1 - there's no place like home");
  setTimeout(recursive,15000);
}
recursive();

```

Now at this setup, it creates one log file every minute without deleting the older ones. Am I doing anything wrong here?

**Comment by mattberther** [comment]
@swateek I looked at the code and ran your example locally and this is what Ive found: the maxFile and maxsize property work together -- in your example, it creates a new log file every minute because the maxsize is not reached during that minute.

Your example would create a `debug.log.yyyy-MM-ddTHH-mm` file and pushes log information into it every 15 seconds. Only if (based on your maxsize setting) you push in more than 104857600 bytes of data during that minute will the log rotate and create a `debug.log.yyyy-MM-ddTHH-mm.1` file. And then, if you push in 104857600 bytes of data into the new file, it would create `debug.log.yyyy-MM-ddTHH-mm.2`. At the time it creates the .2 file, it would delete the first one.

So, in short, to make your example work properly, drop the maxsize WAY down (I set mine at 1024) and then adjust the recursive function to hit more often. I set mine to be 1500ms. When you do this, you can watch the log rotation working properly with maxFiles and maxsize.

**Comment by swateek** [comment]
Matt, I was looking at a different property. 

Let's say I roll the logs everyday. I would want to keep just logs of last 3 days. Is there a way to do it with Winston?

Thank you for the explanation, helped me understand a few things better.

**Comment by mattberther** [comment]
I dont see this functionality in the code yet, but it would make a great addition. Ill try and work up a PR. Thanks!

**Comment by swateek** [comment]
Please, that would be so great. Can I request a feature and close this issue?

**Comment by mattberther** [comment]
Yes, let's close this. Go ahead and submit a new issue as a feature request. Thanks!
